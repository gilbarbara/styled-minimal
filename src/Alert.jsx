import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
