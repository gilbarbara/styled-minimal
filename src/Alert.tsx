import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
