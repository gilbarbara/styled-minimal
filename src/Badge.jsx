import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
