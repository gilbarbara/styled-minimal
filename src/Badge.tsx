import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
