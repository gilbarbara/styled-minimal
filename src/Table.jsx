import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { sizesPropTypes } from './utils/system';

import Box, { basePropTypes } from './Box';

const styles = props => {
  const { bordered, borderless, inverted } = props;
  const { borderColors, colors } = getTheme(props, 'table');

  return css`
    background-color: ${colors[inverted ? 'secondary' : 'primary']};
    ${bordered && !borderless
      ? `border: 1px solid ${borderColors[inverted ? 'secondary' : 'primary']};`
      : ''};
    border-collapse: collapse;
    color: ${colors[inverted ? 'primary' : 'secondary']};
    width: 100%;
  `;
};
const stylesCaption = props => {
  const { captionColor, captionPadding } = getTheme(props, 'table');

  return css`
    caption-side: bottom;
    color: ${captionColor};
    font-size: 90%;
    padding-bottom: ${px(captionPadding)};
    padding-top: ${px(captionPadding)};
    text-align: left;
  `;
};
const stylesCellBorder = props => {
  const { bordered, borderless, inverted } = props;
  const { borderColors } = getTheme(props, 'table');

  if (borderless) {
    return '';
  }

  return css`
    ${bordered ? 'border' : 'border-top'}: 1px solid
      ${borderColors[inverted ? 'secondary' : 'primary']};
  `;
};
const stylesHeadBackgroundColor = props => {
  const { head } = props;
  const { headColors } = getTheme(props, 'table');

  return css`
    background-color: ${headColors[head] || 'transparent'};
  `;
};
const stylesHeadCellBorder = props => {
  const { bordered, borderless, inverted } = props;
  const { borderColors } = getTheme(props, 'table');
  const colorProp = borderColors[inverted ? 'secondary' : 'primary'];

  if (borderless) {
    return '';
  }

  return css`
    ${bordered ? `border: 1px solid ${colorProp}` : `border-bottom: 2px solid ${colorProp}`};
    ${bordered && 'border-bottom-width: 2px'};
  `;
};
const stylesHeadColor = props => {
  const { head } = props;
  const { colors } = getTheme(props, 'table');

  if (head) {
    return css`
      color: ${colors[head === 'dark' ? 'primary' : 'secondary']};
    `;
  }

  return '';
};
const stylesPadding = props => {
  const { size } = props;
  const { padding } = getTheme(props, 'table');

  return css`
    padding: ${px(padding[size])};
  `;
};
const stylesStriped = props => {
  const { inverted, striped } = props;
  const { stripedColors } = getTheme(props, 'table');

  if (striped) {
    return css`
      background-color: ${stripedColors[inverted ? 'secondary' : 'primary']};
    `;
  }

  return '';
};

const Table = styled(Box)`
  ${styles};

  th {
    text-align: inherit;
  }

  th,
  td {
    ${stylesCellBorder};
    ${stylesPadding};
    vertical-align: top;
  }

  thead {
    ${stylesHeadBackgroundColor};
    ${stylesHeadColor};
  }

  thead th {
    ${stylesHeadCellBorder};
    vertical-align: bottom;
  }

  tbody tr:nth-of-type(odd) {
    ${stylesStriped};
  }

  caption {
    ${stylesCaption};
  }
`;

Table.displayName = 'Table';

Table.propTypes = {
  as: PropTypes.string,
  bordered: PropTypes.bool,
  borderless: PropTypes.bool,
  children: PropTypes.node.isRequired,
  head: PropTypes.oneOf(['light', 'dark']),
  inverted: PropTypes.bool,
  size: sizesPropTypes,
  striped: PropTypes.bool,
  ...basePropTypes,
};

Table.defaultProps = {
  as: 'table',
  bordered: false,
  borderless: false,
  inverted: false,
  size: 'md',
  striped: false,
};

export default Table;
