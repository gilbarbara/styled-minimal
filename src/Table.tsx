import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { TableProps } from './types';

const stylesCaption = (props: TableProps) => {
  const { captionColor, captionPadding } = getTheme(props, 'table');

  return css`
    caption-side: bottom;
    color: ${captionColor};
    font-size: 90%;
    padding-bottom: ${px(captionPadding)};
    padding-top: ${px(captionPadding)};
    text-align: left;
  `;
};
const stylesCellBorder = (props: TableProps) => {
  const { frame, clean, dark } = props;
  const { borderColors } = getTheme(props, 'table');

  if (clean) {
    return '';
  }

  return css`
    ${frame ? 'border' : 'border-top'}: 1px solid ${borderColors[dark ? 'secondary' : 'primary']};
  `;
};
const stylesHeadBackgroundColor = (props: TableProps) => {
  const { head } = props;
  const { headColors } = getTheme(props, 'table');

  return css`
    background-color: ${headColors[head!] || 'transparent'};
  `;
};
const stylesHeadCellBorder = (props: TableProps) => {
  const { frame, clean, dark } = props;
  const { borderColors } = getTheme(props, 'table');
  const colorProp = borderColors[dark ? 'secondary' : 'primary'];

  if (clean) {
    return '';
  }

  return css`
    ${frame ? `border: 1px solid ${colorProp}` : `border-bottom: 2px solid ${colorProp}`};
    ${frame && 'border-bottom-width: 2px'};
  `;
};
const stylesHeadColor = (props: TableProps) => {
  const { head } = props;
  const { colors } = getTheme(props, 'table');

  if (head !== 'normal') {
    return css`
      color: ${colors[head === 'dark' ? 'primary' : 'secondary']};
    `;
  }

  return '';
};
const stylesPadding = (props: TableProps) => {
  const { size = 'md' } = props;
  const { padding } = getTheme(props, 'table');

  return css`
    padding: ${px(padding[size])};
  `;
};
const stylesStriped = (props: TableProps) => {
  const { dark, striped } = props;
  const { stripedColors } = getTheme(props, 'table');

  if (striped) {
    return css`
      background-color: ${stripedColors[dark ? 'secondary' : 'primary']};
    `;
  }

  return '';
};

export const StyledTable = styled('table')<TableProps>(
  props => {
    const { frame, clean, dark } = props;
    const { borderColors, colors } = getTheme(props, 'table');

    return css`
      ${baseStyles};
      background-color: ${colors[dark ? 'secondary' : 'primary']};
      ${frame && !clean
        ? `border: 1px solid ${borderColors[dark ? 'secondary' : 'primary']};`
        : ''};
      border-collapse: collapse;
      color: ${colors[dark ? 'primary' : 'secondary']};
      width: 100%;

      th {
        text-align: inherit;
      }

      th,
      td {
        ${stylesCellBorder};
        ${stylesPadding};
        vertical-align: top;
      }

      thead {
        ${stylesHeadBackgroundColor};
        ${stylesHeadColor};
      }

      thead th {
        ${stylesHeadCellBorder};
        vertical-align: bottom;
      }

      tbody tr:nth-of-type(odd) {
        ${stylesStriped};
      }

      caption {
        ${stylesCaption};
      }
    `;
  },
  sx,
  system,
);

export const Table = React.forwardRef<HTMLTableElement, TableProps>((props, ref) => (
  <StyledTable ref={ref} {...props} />
));

Table.defaultProps = {
  frame: false,
  clean: false,
  dark: false,
  head: 'normal',
  size: 'md',
  striped: false,
};
