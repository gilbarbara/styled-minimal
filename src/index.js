export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
