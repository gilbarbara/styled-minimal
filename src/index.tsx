export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
