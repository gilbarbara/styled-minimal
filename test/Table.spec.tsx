import * as React from 'react';
import { render } from '@testing-library/react';

import { Table } from '../src/Table';

const Fixture = props => (
  <Table {...props}>
    <caption>List of users</caption>
    <thead>
      <tr>
        <th scope="col">#</th>
        <th scope="col">First</th>
        <th scope="col">Last</th>
        <th scope="col">Handle</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th scope="row">1</th>
        <td>Mark</td>
        <td>Otto</td>
        <td>@mdo</td>
      </tr>
      <tr>
        <th scope="row">2</th>
        <td>Jacob</td>
        <td>Thornton</td>
        <td>@fat</td>
      </tr>
      <tr>
        <th scope="row">3</th>
        <td>Larry</td>
        <td>the Bird</td>
        <td>@twitter</td>
      </tr>
    </tbody>
  </Table>
);

describe('Table', () => {
  it('should render a basic Table', () => {
    const { container } = render(<Fixture head="dark" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a Table with props', () => {
    const { container } = render(<Fixture clean striped />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
