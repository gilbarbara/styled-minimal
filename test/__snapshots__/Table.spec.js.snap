// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #fff;
  border-collapse: collapse;
  color: #000;
  width: 100%;
}

.c0 th {
  text-align: inherit;
}

.c0 th,
.c0 td {
  border-top: 1px solid #ccc;
  padding: 12px;
  vertical-align: top;
}

.c0 thead {
  background-color: transparent;
}

.c0 thead th {
  border-bottom: 2px solid #ccc;
  vertical-align: bottom;
}

.c0 caption {
  caption-side: bottom;
  color: #ccc;
  font-size: 90%;
  padding-bottom: 6px;
  padding-top: 6px;
  text-align: left;
}

<Table
  as="table"
  bordered={false}
  borderless={false}
  inverted={false}
  size="md"
  striped={false}
>
  <table
    className="c0"
    size="md"
  >
    <caption>
      List of users
    </caption>
    <thead>
      <tr>
        <th
          scope="col"
        >
          #
        </th>
        <th
          scope="col"
        >
          First
        </th>
        <th
          scope="col"
        >
          Last
        </th>
        <th
          scope="col"
        >
          Handle
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          Mark
        </td>
        <td>
          Otto
        </td>
        <td>
          @mdo
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td>
          Jacob
        </td>
        <td>
          Thornton
        </td>
        <td>
          @fat
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          3
        </th>
        <td>
          Larry
        </td>
        <td>
          the Bird
        </td>
        <td>
          @twitter
        </td>
      </tr>
    </tbody>
  </table>
</Table>
`;
